src/shadowbox/server/shared_metrics.ts

```diff
@@ -82,18 +94,61 @@ export class PrometheusUsageMetrics implements UsageMetrics {
 
   constructor(private prometheusClient: PrometheusClient) {}
 
+  private async queryUsage(
+    timeSeriesSelector: string,
+    deltaSecs: number
+  ): Promise<PrometheusQueryResult> {
+    const query = `
+      sum(increase(${timeSeriesSelector}[${deltaSecs}s]))
```

Comment:
The introduction of the string parameter makes this no longer safe. The selector should be escaped. However this code is ok because this method is private and is only called with the two fixed strings below, but please add a comment.